src/transaction/log_page_buffer.c

```diff
@@ -3361,6 +3364,250 @@ logpb_write_toflush_pages_to_archive (THREAD_ENTRY * thread_p)
+      log_Gl.append.prev_lsa = log_Gl.hdr.append_lsa;
+      logpb_next_append_page (thread_p, LOG_SET_DIRTY);
+    }
+  else if (log_Gl.append.log_pgptr->hdr.logical_pageid == log_pgptr->hdr.logical_pageid)
+    {
+      // For the first page, just overwrite the page.
+      // DON'T change meta data like log_Gl.append.prev_lsa. They don't go backward
+    }
+  else
+    {
+      logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "logpb_catchup_append_page");
+    }
+
+  memcpy (log_Gl.append.log_pgptr, log_pgptr, LOG_PAGESIZE);
+  logpb_set_dirty (thread_p, log_Gl.append.log_pgptr);
+
+  logpb_catchup_end_append (thread_p, log_pgptr);
+}
+
+/*
+ * logpb_catchup_start_append - Start appending a page during catch-up.
+ *      LOGPB_APPENDREC_SUCCESS: 
+ *        A normal case. The previous appending has been successfully.
+ *      LOGPB_APPENDREC_IN_PROGRESS:
+ *        The previous appended page doesn't have a start of a record. It only constains a middle part 
+ *        or the last part of a record. In this case, the prev_lsa has not been advanced, 
+ *        and the EOF will be appended when flushing log pages halfway through appending a page.
+ *      LOGPB_APPENDREC_PARTIAL_FLUSHED_END_OF_LOG:
+ *        Log pages are flushed during appending a page including only a partial record, and the temporary EOF 
+ *        has been appended at the prev_lsa. We will remove this at logpb_catchup_end_append() when the partial record completes.
+ *      
+ *      The status transition:
+ *   
+ *       ┌──► ┌─►_SUCCESS
+ *       │    │      │
+ *       │    │ (*)  │ start appending a page
+ *       │    │      ▼
+ *       │    └──_IN_PROGRESS
+ *       │           │
+ *       │           │ It happens to flush log pages during the catch-up.
+ *       │           │ so EOF is appended at the prev_lsa.
+ *       │           ▼
+ *       │       _PARTIAL_FLUSHED_END_OF_LOG
+ *       │           │
+ *       │           │ It's done to append log pages containing an incomplete log record.
+ *       │           │ If there is no incomplete log record, just a page has been appended.
+ *       │           ▼
+ *       │       _PARTIAL_ENDED
+ *       │           │
+ *       │           │ flush log pages to remove the EOF.
+ *       │           │ It appends the EOF at the append_lsa instead.
+ *       │           ▼
+ *       │       _PARTIAL_FLUSHED_ORIGINAL
+ *       │           │
+ *       │           │ Remove the EOF at the append_lsa on the memory buffer.
+ *       └───────────┘ This will be flushed in the next time.
+ *
+ * (*): It's done to append log pages containing an incomplete log record without flushing in the middle.
+ * 
+ *      See logpb_catchup_end_append() and logpb_flush_all_append_pages() as well for the complete picture.
+ *     
+ *   log_pgptr(in): a log page pointer to append
+ */
+static void
+logpb_catchup_start_append (THREAD_ENTRY * thread_p)
+{
+  assert (LOG_CS_OWN_WRITE_MODE (thread_p));
```

Comment:
I did it intentionally as other does like `logpb_prior_lsa_append_all_list(`) -> `logpb_append_prior_lsa_list()`. I think it's worth it in case the `logpb_catchup_append_page` is changed or this function is used in others.